### PR TITLE
[REVIEW] All null string entries should have null data buffer

### PR DESCRIPTION
**Pull request by kuhushukla**
Passing a one byte data buffer for all null strings case can cause corruption given that the cudf layer can handle null data buffers. This PR address this issue by letting the buffer be null if there is no data.

### CODE REVIEW

**Comment by kuhushukla** [comment]
My familiarity with this code is limited and I could use some comments on any case I have inadvertently missed @revans2 . Tests pass for the case I was seeing a one byte overflow problem without this patch.

File: java/src/main/java/ai/rapids/cudf/JCudfSerialization.java [hunk 1]
```diff
@@ -1037,7 +1037,11 @@ private static long copySlicedOffsets(DataWriter out, ColumnBufferProvider colum
       // Don't copy anything, there are no rows
       return 0;
     }
-    long bytesToCopy = (numRows + 1) * 4;
+    long bytesToCopy = 0;
+    if (column.getHostBufferFor(ColumnVector.BufferType.DATA) != null) {
+      bytesToCopy = (numRows + 1) * 4;
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/JCudfSerialization.java]: This is adding in the number of bytes needed to copy the offsets.  That will never change for a string.  We are only looking at the data portion, which was already not being copied.

File: java/src/main/java/ai/rapids/cudf/JCudfSerialization.java [hunk 2]
```diff
@@ -1294,11 +1298,7 @@ private static Table sliceUpColumnVectors(SerializedTableHeader header,
             offsets = combinedBuffer.slice(offsetInfo.offsets, offsetInfo.offsetsLen);
           }
 
-          if (offsetInfo.dataLen == 0) {
-            // The vector is possibly full of null strings. This is a rare corner case, but here is the
-            // simplest place to work around it.
-            data = DeviceMemoryBuffer.allocate(1);
-          } else {
+          if (offsetInfo.dataLen > 0) {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/JCudfSerialization.java]: I would like to keep a comment about why the data buffer can be empty.

**Comment by revans2** [comment]
Also please remove the WIP and add in the PR to the changelog.

File: java/src/main/java/ai/rapids/cudf/JCudfSerialization.java (hunk 1 above)
Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/JCudfSerialization.java]: I see. Thanks for pointing out. I thought my test needed this due to a failure to get array length later in sliceAndPad but I verified that my test was not in any need for this change. Removed.

**Comment by kuhushukla** [comment]
I am still seeing (unrelated) CI failures after merging. Is there something we can do to unblock this? @revans2 , @kkraus14 . Thank you for any inputs on this.

**Comment by kkraus14** [comment]
@revans2 @kuhushukla given this is Java it isn't tested by CI anyway so I'm happy to admin merge if you want. Does that work for you?

**Comment by kuhushukla** [comment]
> @revans2 @kuhushukla given this is Java it isn't tested by CI anyway so I'm happy to admin merge if you want. Does that work for you?

Yes, that would work and thanks a lot @kkraus14  
(Added a removed newline and anything relevant to us on the CI looks fine to me)

**Pull request merged**